File: local/HDR__/cpp/cpp/shape/shape.cpp
#include<cmath>
#include "shape.h"

Shape::Shape(int x, int y)
: x_(x), y_(y)
{

}

void Shape::move(int offsetX, int offsetY){
    y_ += offsetY;
    x_ += offsetX;
}

double Rectangle::getDiagonal()const{
    return sqrt(width_ * width_ + height_ * height_);
}

// area() ?? 

File: local/HDR__/python/shape/shape/shape.py
#shape.py
from abc import *

class Shape(metaclass=ABCMeta):
    def __init__(self, x = 0, y = 0):
        self.__x = x
        self.__y = y
        
    def move(self, offset_x, offset_y):
        self.__x += offset_x
        self.__y += offset_y
    
    @abstractmethod 
    def area():
        pass


